nncf/experimental/post_training/graph/model_transformer.py

```diff
@@ -49,24 +49,14 @@ def prepare_model_for_statistics_collection(self, statistic_points: StatisticPoi
         transformation_layout = self._get_transformation_layout_extra_outputs(statistic_points)
         return self.transform(transformation_layout)
 
+    @abstractmethod
     def transform(self, transformation_layout: TransformationLayout) -> ModelType:
```

Comment:
IMHO, I see no reason to have the StaticModelTransformerBase entity.

Reply:
But what about the methods like `set_model` and `prepare_model_for_statistics_collection`? And others methods for each transformation. Where should we place them instead of `StaticModelTransformerBase`?

Reply:
> But what about the methods like ```set_model``` and ```prepare_model_for_statistics_collection```?

- Could you briefly explain which scenario requires the ``set_model``` function?
- The ```prepare_model_for_statistics_collection``` has the following flow:
``` model -> statistics_points -> transformation_layout -> transform -> model```
It looks like creating transformation_layout by statistics point is algorithm's responsibility, because the method of collecting statistics can be depends on algorithm's parameters (graph-based method, collection of outputs)...